src/module/actor/mixins/actor-conditions.js

```diff
@@ -66,14 +92,50 @@ export const ActorConditionsMixin = (superclass) => class extends superclass {
             if (conditionItem) {
                 const promise = this.deleteEmbeddedDocuments("Item", [conditionItem.id]);
                 promise.then(() => {
-                    const updateData = {};
-                    updateData[`data.conditions.${conditionName}`] = false;
-                    this.update(updateData).then(() => {
-                        Hooks.callAll("onActorSetCondition", {actor: this, item: conditionItem, conditionName: conditionName, enabled: enabled});
-                    });
+                    this._updateActor(conditionName, false);
+                    Hooks.callAll("onActorSetCondition", {actor: this, item: conditionItem, conditionName: conditionName, enabled: enabled});
```

Comment:
Why did you take the onActorSetCondition callback out of the actor update? It was done on purpose so that the callback would not fire until the actor was fully finished updating.

Reply:
same as above